src/Tokenizer/Tokens.php

```diff
@@ -1366,7 +1454,7 @@ private function unregisterFoundToken($token): void
     }
 
     /**
-     * @param array{int}|string|Token $token token prototype
+     * @param array{0: int, 1?: string}|string|Token $token token prototype
```

Comment:
Why all these changes from `array{int}` to `array{0: int, 1?: string}`? These functions ignore the `1` index and the new signature suggests otherwise.

Reply:
Yeah I could have skipped this one. Althought I think it would be nice to introduce a type alias somewhere (e.g. `TokenPrototype`) and use it to replace array shapes. Having consistent declarations would help with that.